pkg/registry/core/service/portallocator/metrics.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package portallocator
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "kube_apiserver"
+	subsystem = "nodeport_allocator"
+)
+
+var (
+	// nodePortAllocated indicates the amount of ports allocated by NodePort Service.
+	nodePortAllocated = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "allocated_ports",
+			Help:           "Gauge measuring the number of allocated NodePorts for Services",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"cidr"},
```

Comment:
you don't need  a vector in all this metrics, the NodePorts are global, there is no range per ip family

Reply:
you are right, It's done